webapp/src/main/webapp/iaf/frankdoc/js/app.js

```diff
@@ -102,6 +102,16 @@ angular.module('iaf.frankdoc').config(['$stateProvider', '$urlRouterProvider', f
 				if(searchTextLC) {
 					if(JSON.stringify(obj).replace(/"/g, '').toLowerCase().indexOf(searchTextLC) > -1) {
 						r[element] = obj;
+					} else { // search in parent
+						let elementParent = elements[element].parent;
+						while(elementParent){
+							let parentObj = elements[elementParent];
+							if(JSON.stringify(parentObj).replace(/"/g, '').toLowerCase().indexOf(searchTextLC) > -1) {
+								r[element] = obj;
+								break;
+							}
+							elementParent = elements[elementParent].parent;
+						}
```

Comment:
Right now this loops through all elements, and then all parent elements of each element recursively. This makes this loop very inefficient, and very complex.

It would be better to loop through the `$scope.elements` once and filter out which elements should be returned.